cpp/models/abm/infection.cpp

```diff
@@ -100,86 +104,94 @@ bool Infection::is_detected() const
     return m_detected;
 }
 
+TimePoint Infection::get_init_date() const
+{
+    return m_viral_load.start_date;
+}
 TimePoint Infection::draw_infection_course(AgeGroup age, const GlobalInfectionParameters& params, TimePoint init_date,
-                                           InfectionState init_state)
+                                           InfectionState init_state,
+                                           std::pair<ProtectionType, TimePoint> latest_protection)
 {
     TimePoint start_date = draw_infection_course_backward(age, params, init_date, init_state);
-    draw_infection_course_forward(age, params, init_date, init_state);
+    draw_infection_course_forward(age, params, init_date, init_state, latest_protection);
     return start_date;
 }
 
 void Infection::draw_infection_course_forward(AgeGroup age, const GlobalInfectionParameters& params,
-                                              TimePoint init_date, InfectionState start_state)
+                                              TimePoint init_date, InfectionState start_state,
+                                              std::pair<ProtectionType, TimePoint> latest_protection)
 {
     auto t = init_date;
     TimeSpan time_period{}; // time period for current infection state
     InfectionState next_state{start_state}; // next state to enter
     m_infection_course.push_back(std::pair<TimePoint, InfectionState>(t, next_state));
-    auto uniform_dist = UniformDistribution<double>::get_instance();
+    auto uniform_dist                     = UniformDistribution<double>::get_instance();
+    ScalarType severity_protection_factor = 0.5;
```

Comment:
For the switch statement, all variables have to be defined in advance. Otherwise, I received an error `cannot jump from switch statement to this case label`

Reply:
You can create a scope for the switch cases with brackets '{}' around the case such that you then can define new variables. It's not a big deal though. :)

Reply:
Ok, it's done!